src/avram/associations/has_many.cr

```diff
@@ -1,8 +1,14 @@
 module Avram::Associations::HasMany
-  macro has_many(type_declaration, through = nil, foreign_key = nil)
+  macro has_many(type_declaration, through = nil, source = nil, foreign_key = nil)
```

Comment:
Source is a property that can be passed to dictate the exact association we look for on the connecting model. If it is not specified we singularize the association name. That means that source is required if you want to create a has_many through a has_many relationship because the relationship name would be plural.

I can't think of a good way to automatically handle calling the correct methods on a has_many AND a belongs_to. I also can't find any examples of a compiled, type-safe, Rails-like ORM to learn from.